compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -6146,6 +6146,74 @@ TR::Register *commonLoadEvaluator(TR::Node *node, TR::InstOpCode::Mnemonic op, i
    return commonLoadEvaluator(node, op, size, tempReg, cg);
    }
 
+static TR::Register *
+loadAddress(TR::Node *node, TR::MemoryReference *mr, TR::CodeGenerator *cg)
+   {
+   TR::Register *targetReg = cg->allocateRegister();
+
+   // TODO: this probably exists somewhere else
+   TR::Register *baseReg = mr->getBaseRegister();
+   TR::Register *indexReg = mr->getIndexRegister();
+   intptr_t offset = mr->getOffset();
+
+   if (offset == 0)
+      {
+      if (baseReg != NULL && indexReg == NULL)
+         {
+         if (baseReg != targetReg)
+            generateMovInstruction(cg, node, targetReg, baseReg);
+         }
+      else if (indexReg != NULL && baseReg == NULL)
+         {
+         if (baseReg != targetReg)
+            generateMovInstruction(cg, node, targetReg, indexReg);
+         }
+      else
+         {
+         generateTrg1Src2Instruction(cg, TR::InstOpCode::addx, node, targetReg, baseReg, indexReg);
+         }
+      }
+   else
+      {
+      if (baseReg == NULL)
+         {
+         generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addimmx, node, targetReg, indexReg, offset);
+         }
+      else if (indexReg == NULL)
+         {
+         generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addimmx, node, targetReg, baseReg, offset);
+         }
+      else
+         {
+         generateTrg1Src2Instruction(cg, TR::InstOpCode::addx, node, targetReg, baseReg, indexReg);
+         generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addimmx, node, targetReg, targetReg, offset);
+         }
```

Comment:
AArch64 has pre-index and post-index variants of LDR (immediate)/STR (immediate) instructions, like `strprew` and `vldrpostq` in our notation.
You need to update the address in `baseReg` before or after the memory access.
In addition, you must not add the `offset` in calculating the target address for the post-index variants.